collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -320,4 +320,98 @@ abstract class EntitiesRepositoryTest {
         assertThat(wines[1].index, equalTo(1))
         assertThat(wines[2].index, equalTo(2))
     }
+
+    @Test
+    fun `#getById returns entities with matching id`() {
+        val repository = buildSubject()
+
+        val leoville = Entity.New("wines", "1", "Léoville Barton 2008")
+        val canet = Entity.New("wines", "2", "Pontet-Canet 2014")
+        repository.save(leoville, canet)
+
+        val wines = repository.getEntities("wines")
+
+        val queriedLeoville = repository.getById("wines", "1")
+        assertThat(queriedLeoville, equalTo(wines.first { it.id == "1" }))
```

Comment:
We want to check that we're getting the same `Entity.Saved` (which has the extra `index` property currently) back as we would for a `getEntities` call so that wouldn't be enough.